### PR TITLE
Feature/loop syntax

### PR DESCRIPTION
**Pull request by wesen**
Out of the same clarity/optimization needs, I decided to use {{, }} to iterate (and only iterate) over an array (or a hash depending on the language), rather than the "i can do everything" {{#section}}.

let me know what you think

### CODE REVIEW

File: specs/loop.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  - name: Nested loops single
+    desc: Loop over a hash should not push a new context
+    data:
+      a: 'first'
+      array:
+        a: 1
+        b: 2
+        c: 3
+    template: '{{,array}}{{a}} {{/array}}'
+    expected: 'first first first '
```
Comment (urzds) [inline specs/loop.yml]: This test looks exactly like the previous one. Where is the difference?

**Comment by urzds** [comment]
I also support this request, as I would like to iterate over a hash (key/value pairs). Could you please add an example of how that would look like? I propose `{{,hash}}key: {{key}}, value: {{value.nstd.idx}}{{/hash}}`

Using comma `,` as the "keyword" seems a bit unintuitive, but other than `*` I also have no good idea what else to use.

**Comment by urzds** [comment]
What was the reason for closing this?

**Comment by wesen** [comment]
Just closing after 5 years to clean up my open pull request view :)

**Comment by urzds** [comment]
Shall I submit the patch in a new pull request?

**Pull request closed without merging**